tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -265,3 +265,25 @@ TEST(SGSparseMatrix, from_dense)
 		for (index_t vecIndex=0; vecIndex<numberOfVectors; ++vecIndex)
 			EXPECT_EQ(sparseMatrix(featIndex,vecIndex), denseMatrix(featIndex,vecIndex));
 }
+
+TEST(SGSparseMatrix, test_transposed_square_matrix)
```

Comment:
@karlnapf Btw, all the other tests do not implement the style fixes you suggested (about camel case and the for loops). You can check.